parsers/ES.py

```diff
@@ -0,0 +1,351 @@
+#!/usr/bin/env python3
+
+import datetime
+import logging
+
+# The arrow library is used to handle datetimes
+from arrow import get
+
+# package "ree" is used to parse data from www.ree.es
+# maintained on github by @hectorespert at https://github.com/hectorespert/ree
+from ree import (
+    BalearicIslands,
+    ElHierro,
+    Formentera,
+    Gomera,
+    GranCanaria,
+    Ibiza,
+    LanzaroteFuerteventura,
+    LaPalma,
+    Mallorca,
+    Menorca,
+    Tenerife,
+)
+
+# The request library is used to fetch content through HTTP
+from requests import Session
+
+from .lib.exceptions import ParserException
+from .lib.validation import validate
+
+# Minimum valid zone demand. This is used to eliminate some cases
+# where generation for one or more modes is obviously missing.
+FLOORS: dict[str, int] = {
+    "ES-CN-FVLZ": 50,
+    "ES-CN-GC": 150,
+    "ES-CN-IG": 3,
+    "ES-CN-LP": 10,
+    "ES-CN-TE": 150,
+    "ES-CN-HI": 2,
+    ## Guess we'll need to figure these out later?! Adapted from ES-CN:
+    "ES-IB": 0,
+    "ES-IB-FO": 0,
+    "ES-IB-IZ": 0,
+    "ES-IB-MA": 0,
+    "ES-IB-ME": 0,
+}
+
+
+def fetch_island_data(zone_key: str, session):
+    if zone_key == "ES-CN-FVLZ":
+        lanzarote_fuerteventura_data = LanzaroteFuerteventura(session).get_all()
+        if not lanzarote_fuerteventura_data:
+            raise ParserException(zone_key, "LanzaroteFuerteventura not response")
+        else:
+            return lanzarote_fuerteventura_data
```

Comment:
Yeah I can do that, right now this is just a copy paste from the original parsers.

PS: Now you know why I wanted to use match case (so I could keep the original structure) 😉

Reply:
Haha yeah, I see - but I think a mapping would be even better than that match case anyway 😄

Reply:
I have updated the parser with your suggestion (opted for early return of data and the exception in the else block though).

In regards to better handling non existent zones I'll do some digging if we can use type literals to make sure the `zone_key` is always valid, otherwise we can just define a list and check if the key is in that (or reuse the keys from the function or floor dict perhaps?)